s2/bq/doc/REFERENCE.md

```diff
@@ -0,0 +1,162 @@
+## Reference
+
+### S2
+
+#### VERSION
+
+{{% bannerNote type="code" %}}
+s2.VERSION()
+{{%/ bannerNote %}}
+
+Returns the current version of the s2 library.
+
+#### GEOJSONBOUNDARY_FROMKEY
+
+{{% bannerNote type="code" %}}
+s2.GEOJSONBOUNDARY_FROMKEY(key STRING) 
+{{%/ bannerNote %}}
+
+Returns a GeoJSON containing the boundary corners of a quadkey.
+
+* `key`: `STRING` quadkey to extract the corners from.
+
+#### GEOJSONBOUNDARY_FROMLONGLAT
+
+{{% bannerNote type="code" %}}
+s2.GEOJSONBOUNDARY_FROMLONGLAT(longitude FLOAT64, latitude FLOAT64, level NUMERIC)
+{{%/ bannerNote %}}
+
+Returns a GeoJSON containing the boundary corners given a longitude, latitude and zoom level.
+
+* `longitude`: `FLOAT64` horizontal coordinate on the map.
+* `latitude`: `FLOAT64` vertical coordinate on the map.
+* `level`: `NUMERIC` Level of detail or zoom.
+
+#### ID_FROMKEY
+
+{{% bannerNote type="code" %}}
+s2.ID_FROMKEY(key STRING)
+{{%/ bannerNote %}}
+
+Convert from hilbert quadtree id to s2 cell id.
+
+* `key`: `STRING` quadkey to be converted.
+
+#### ID_FROMLONGLAT
+
+{{% bannerNote type="code" %}}
+s2.ID_FROMLONGLAT(longitude FLOAT64, latitude FLOAT64, level NUMERIC)
+{{%/ bannerNote %}}
+
+Generate the s2 cell id for a given longitude, latitude and zoom level.
+
+* `longitude`: `FLOAT64` horizontal coordinate on the map.
+* `latitude`: `FLOAT64` vertical coordinate on the map.
+* `level`: `NUMERIC` Level of detail or zoom.
+
+#### KEY_FROMID
+
+{{% bannerNote type="code" %}}
+s2.KEY_FROMID(id INT64)
+{{%/ bannerNote %}}
+
+Convert from s2 cell id to hilbert quadtree id.
+
+* `id`: `INT64` s2 cell id to be converted.
+
+#### KEY_FROMLONGLAT
+
+{{% bannerNote type="code" %}}
+s2.KEY_FROMLONGLAT(longitude FLOAT64, latitude FLOAT64, level NUMERIC)
+{{%/ bannerNote %}}
+
+Generate the quadkey for a given longitude, latitude and zoom level.
+
+* `longitude`: `FLOAT64` horizontal coordinate on the map.
+* `latitude`: `FLOAT64` vertical coordinate on the map.
+* `level`: `NUMERIC` Level of detail or zoom.
+
+#### LONG_FROMID
```

Comment:
What's the reasoning for this function? As @SpencerC mentions, the fact that this is returning the latitude of the centroid is far from obvious and I'm not sure if we want to provide that.

A good start would be to have a function that returns the geography that represents the cell, then if people want the centroid or the latitude of the centroid they can get it using BigQuery functions. If we saw that everybody was using this for their visualizations (God forbit them) then we could think of adding a helper function (again, using the battle tested BigQuery functions), but as far as I can see currently it doesn't make much sense to provide this.